PipelinesTasks/PRMetrics/buildAndReleaseTask/src/metrics/codeMetrics.ts

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { FixedLengthArray } from '../utilities/fixedLengthArray'
+import { ICodeFileMetric } from './iCodeFileMetric'
+import { singleton } from 'tsyringe'
+import * as taskLib from 'azure-pipelines-task-lib/task'
+import CodeMetricsData from './codeMetricsData'
+import GitInvoker from '../git/gitInvoker'
+import Inputs from './inputs'
+import TaskLibWrapper from '../wrappers/taskLibWrapper'
+
+/**
+ * A class for computing metrics for software code in pull requests.
+ */
+@singleton()
+export default class CodeMetrics {
+  private _gitInvoker: GitInvoker
+  private _inputs: Inputs
+  private _taskLibWrapper: TaskLibWrapper;
+
+  private _isInitialized: boolean = false
+  private _ignoredFiles: string[] = []
+  private _size: string = ''
+  private _sizeIndicator: string = ''
+  private _metrics: CodeMetricsData = new CodeMetricsData(0, 0, 0)
+  private _isSufficientlyTested: boolean | null = null
+
+  /**
+   * Initializes a new instance of the `CodeMetrics` class.
+   * @param gitInvoker The Git invoker.
+   * @param inputs The inputs passed to the task.
+   * @param taskLibWrapper The wrapper around the Azure Pipelines Task Lib.
+   */
+  constructor (gitInvoker: GitInvoker, inputs: Inputs, taskLibWrapper: TaskLibWrapper) {
+    this._gitInvoker = gitInvoker
+    this._inputs = inputs
+    this._taskLibWrapper = taskLibWrapper
+  }
+
+  /**
+   * Gets the collection of ignored files.
+   * @returns The collection of ignored files.
+   */
+  public get ignoredFiles (): string[] {
+    this._taskLibWrapper.debug('* CodeMetrics.ignoredFiles')
+
+    this.initialize()
+    return this._ignoredFiles
+  }
+
+  /**
+   * Gets the size.
```

Comment:
do you think comments like this could be a bit more descriptive? The method signature is pretty self-explanatory, or are the comments used more for documentation purposes?